sync/downstream.py

```diff
@@ -811,6 +812,47 @@ def update_metadata(self, log_files, stability=False):
 
         return disabled
 
+    @mut()
+    def get_gh_metadata(self):
+        """
+        Retrieve the metadata from the upstream PR status checks. This will try to find the
+        Taskcluster run associated with this PR and download its logs.
+
+        :return: True if successful, False otherwise
+        """
+        pr_id = self.pr
+        _, statuses = env.gh_wpt.get_combined_status(pr_id)
+        for status in statuses:
+            if status.context == env.config['web-platform-tests']['ci']['context']:
+                taskgroup_id = status.target_url.rsplit('/', 1)[1]
+                taskgroup = TaskGroup(taskgroup_id)
+                task_name = 'wpt-firefox-nightly-results'
+
+                def t_filter(x):
+                    return x['task']['metadata']['name'] == task_name
+
+                tasks = taskgroup.view(filter_fn=t_filter)
+
+                if len(tasks) != 1:
+                    logger.error("Could not find the TaskCluster task for %s" % task_name)
+                    return False
+
+                trypush.download_upstream_logs(self, tasks)
+
+                try:
+                    # TODO replace lines when update_metadata enabled
+                    # log_path =tasks.tasks[0]['status']['runs'][0]['_log_paths']['wpt_report.json']
+                    tasks.tasks[0]['status']['runs'][0]['_log_paths']['wpt_report.json']
+                except KeyError:
+                    logger.warning("Log path not found for downloaded logs from PR Taskcluster run")
+                    return False
+                # TODO Disabled till we are sure it is working correctly
+                # self.update_metadata([log_path])
```

Comment:
I think in any case this call should be outside this code.

Reply:
Agreed, I think it will be better once the try and upstream CI process is merged. Will remove for now and we can figure out where it will go in the future